sshbuf-misc.c

```diff
@@ -279,6 +285,11 @@ sshbuf_read(int fd, struct sshbuf *buf, size_t maxlen, size_t *rlen)
 	ssize_t rr;
 	u_char *d;
 
+	if(d == NULL)
+	{
+		printf("null found");
```

Comment:
you can't just send output to stdout in from the deamon as it probably doesn't have a stdout (or worse, fd 1 is used for something important).